packages/base-driver/lib/basedriver/capabilities.js

```diff
@@ -336,43 +335,74 @@ function processCapabilities(
 
 /**
  * Return a copy of a capabilities object which has taken everything within the 'options'
- * capability and promoted it to the top level. Note that this function is assumed to be run after
- * all vendor prefixes have already been stripped from the top level. So we are dealing with e.g.
- * 'options' and not 'appium:options' at this point. Any prefixes _inside_ the 'options' capability
- * will themselves be stripped. This is designed as an internal function, not one to operate on
- * user-constructed capabilities.
+ * capability and promoted it to the top level.
  *
- * @param {object} originalCaps - the capabilities to analyze and promote from 'options'
- * @return {object!} - the capabilities with 'options' promoted if necessary
+ * @template {Constraints} C
+ * @param {import('@appium/types').W3CCapabilities<C>} originalCaps
+ * @return {import('@appium/types').W3CCapabilities<C>} the capabilities with 'options' promoted if necessary
  */
 function promoteAppiumOptions(originalCaps) {
-  const appiumOptions = originalCaps[APPIUM_OPTS_CAP];
-  if (!appiumOptions) {
-    return originalCaps;
-  }
+  const promoteForObject = (obj) => {
+    const appiumOptions = obj[PREFIXED_APPIUM_OPTS_CAP];
+    if (!appiumOptions) {
+      return obj;
+    }
```

Comment:
```suggestion
    if (_.isEmpty(appiumOptions)) {
      return obj;
    }
```

.. then delete L354-356, since it's redundant.

Reply:
isEmpty also returns true for an empty object (`{}`) I don't want it to be the case here